### DISCUSSION TITLE
Loading custom embedding model besides those in sentence-transformers library

### DISCUSSION THREAD

**Discussion by micnikolaev** [open-webui/open-webui#1535] [General]
Right now it is not possible to load intfloat/multilingual-e5-large-instruct because the embedding model is loaded via SentenceTransformerEmbeddingFunction which only load libraries in https://huggingface.co/sentence-transformers. Can we make it possible to load  intfloat/multilingual-e5-large-instruct in the future?

**Comment by ghost** [comment]
Hi, this should be supported on the latest release with the proper syntax. If a model isn't a _sentence-transformers_ release, short-name isn't available and you need to include the Hugging Face prefix. However it must be an ["SBERT" compatible model](https://www.sbert.net/docs/pretrained_models.html#model-overview) categorized as a [sentence-transformer](https://huggingface.co/models?library=sentence-transformers). For example:

- Building: `--build-arg="USE_EMBEDDING_MODEL=mixedbread-ai/mxbai-embed-large-v1"`
- Runtime environment variable: `USE_EMBEDDING_MODEL=mixedbread-ai/mxbai-embed-large-v1`
- GUI addition: **mixedbread-ai**/mxbai-embed-large-v1

Try using exactly `intfloat/multilingual-e5-large-instruct`. I've just tested it with CUDA and it works great.

![image](https://github.com/open-webui/open-webui/assets/126992880/77c5e013-d555-4f04-b8f5-ea42d78273cb)

**Comment by Hugobox** [comment]
Hi! I tried this model in the latest openWebUI v.0.3.1 but while everything seems to work (aka no errors) when prompting against a document, it always returns something along the lines of "context is empty". Then I try with the default model sentence-transformers/all-MiniLM-L6-v2 and it works (but only if the prompt and the document are in english)

**Comment by Mhnd3** [comment]
i am facing the same issue as @Hugobox, using any model instead of the default one the context is just empty

**Comment by viba1** [comment]
Same problem here, using Open WebUI 0.3.16. Trying to change engine to "ollama" (0.3.8) and using a embedding model leads to "context empty", even after having reload the document.

**Reply by Vinay-052** [reply]
same issue here. Tried out multiple models from huggingface including from sentence transformers and it search within docs doesnt work.

**Comment by wojciech-dron** [comment]
You need to click on uploaded file and select entire document:
![image](https://github.com/user-attachments/assets/537cf560-10dd-4fd7-a725-c0595fe117d5)

**Reply by MrStrategy** [reply]
If you do so you actually skip the embeddings engine. This may be OK for smaller documents but not for larger ones or even a knowledge base

**Reply by rickCSMF** [reply]
This works. But is there a way to set Using Entire Document as the default? Some users won't know to go into the upload and set this. (Thus, the reason this is most likely being reported). It's not as intuitive as the rest of the UI of openwebUI.
